### ISSUE TITLE
gradient mixin vertical-three-colors on <tr> doesn't render correctly on IE11 or Edge

### ISSUE THREAD

**Issue by RodolpheGohard** [twbs/bootstrap#18504]
Hi,

when applying a vertical-three-colors gradient on a `<tr>` in IE11, I get a buggy display: the gradient does'nt extend to the full width of the `<tr>`.

Here is a demo:

```html
<style type="text/css">
   /* GENERATED FROM LESS .gradiented {
    .vertical-three-colors(#cccccc, #e9e9e9, 5%, #e9e9e9);
    }*/
    .gradiented {
        background-image: -webkit-linear-gradient(#cccccc, #e9e9e9 5%, #e9e9e9);
        background-image: -o-linear-gradient(#cccccc, #e9e9e9 5%, #e9e9e9);
        background-image: linear-gradient(#cccccc, #e9e9e9 5%, #e9e9e9);
        background-repeat: no-repeat;
        filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#ffcccccc', endColorstr='#ffe9e9e9', GradientType=0);
    }
</style> 

<table>
    <tr>
        <td>cell 1</td>
        <td>cell 2</td>
        <td>cell 3</td>
        <td>cell 4</td>
    </tr>
    <tr class="gradiented">
        <td>cell 1</td>
        <td>cell 2</td>
        <td>cell 3</td>
        <td>cell 4</td>
    </tr>
</table>

```

in this sample, running IE11, I only get the first cell to be filled. The situation doesn't happen if I remove the ` background-repeat: no-repeat;` property, which makes me wonder why it's there to begin with.

**Comment by cvrebert** [comment]
JS Fiddle: http://jsfiddle.net/cvrebert/1g6dpwu4/1/

**Comment by cvrebert** [comment]
Confirmed in Microsoft Edge:
<img width="418" alt="msft-edge" src="https://cloud.githubusercontent.com/assets/419884/11701621/7206aa86-9e85-11e5-9090-7e6da0a87472.png">

**Comment by cvrebert** [comment]
> The situation doesn't happen if I remove the `background-repeat: no-repeat;` property, which makes me wonder why it's there to begin with.

I'm not enough of a CSS person to know exactly why, but some poking around suggests it's so you get https://jsbin.com/goweruq/edit?html,css,output instead of https://jsbin.com/sihobo/edit?html,css,output
(I assume there are other cases besides `<body>` where this applies.)
CC: @mdo

**Comment by twbs-lmvtfy** [comment]
Hi @cvrebert!

You appear to have posted a live example (https://jsbin.com/goweruq/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by twbs-lmvtfy** [comment]
Hi @cvrebert!

You appear to have posted a live example (https://jsbin.com/sihobo/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
Filed Edge/IE11 bug: https://connect.microsoft.com/IE/feedback/details/2110930/edge-ie11-gradient-background-image-on-tr-is-only-applied-to-first-td
Assuming the `background-repeat: no-repeat` turns out to be intentional, there's nothing further we can do. It's a bug in the browser itself.

**Comment by RodolpheGohard** [comment]
thanks for the investigation. I guess I'll have to force `background-repeat: no-repeat`, then.

Aside of this, I'm a bit surprised by the default repeat behaviour, since we already have repeating-linear-gradient for that purpose. I tried browsing the specs but couldn't figure out where this repeating gradient behaviour comes from.

**Comment by RodolpheGohard** [comment]
I've a suggestion: maybe we could set `background-repeat: repeat-x` instead of `no-repeat` for vertical gradients ? tried it with the examples you provided and it seems to give fine results.

**Comment by RodolpheGohard** [comment]
I can make https://github.com/RodolpheGohard/bootstrap/commit/36dc4119ba0fbec79623959457f2f8000d7a675a into a pull request if needed, not sure if I should include all the dist files though.

**Comment by cvrebert** [comment]
Pull requests shouldn't include dist files. But I'd advise waiting for @mdo to reply before opening the pull request.

**Comment by mdo** [comment]
[**Bootstrap 3 is no longer being officially developed or supported.**](https://github.com/twbs/bootstrap/issues/20631)

All work has moved onto our next major release, v4. As such, this issue or pull request is being closed as a "won't fix." For additional help and support, we recommend [utilizing our community resources](https://github.com/twbs/bootstrap/blob/master/README.md#community). Thanks for your understanding, and see you on the other side of v4!

<3,
@mdo and team
